src/firmware/shared/circular_buffer.c

```diff
@@ -0,0 +1,54 @@
+#include "firmware/shared/circular_buffer.h"
+
+void circular_buffer_init(struct circular_buffer *cbuff)
```

Comment:
to be consistent with how we have been doing memory allocation elsewhere we should change this to

`CircularBuffer_t circular_buffer_create()`

Reply:
Done following `firmware_world.h` as an example.